cmd/entrypoint/runner.go

```diff
@@ -82,30 +84,70 @@ func (rr *realRunner) Run(ctx context.Context, args ...string) error {
 	signal.Notify(rr.signals)
 	defer signal.Reset()
 
+	var err error
```

Comment:
Could you please add a test for the logic added in this file?

Reply:
done